### PR TITLE
Merge NEI hide Item update 2

### PR DESCRIPTION
**Pull request by SvenKayser**

### CODE REVIEW

**Comment by reteo** [comment]
For the milk bucket fix: do we really need an anonymous function if the code isn't being performed on a multidimensional list?  The main reason the anonymous functions were there was to produce a local copy of the lists the code worked on, thereby reducing overhead from re-reading the global lists over and over.  Since the milk bucket code only handles one item, not running down a multidimensional list, any performance benefit from the anonymous functions should be minimal.  That can be easily fixed after merging, but just in case it's important...

**Comment by SvenKayser** [comment]
Scoping sections of code is merely a habit rather than actually beneficial here, so it's mostly stylistic rather than actually serving a function. It's something you pick up when writing code bits in JavaScript that are supposed to be placed in / loaded into existing code (as you do a lot when doing web programming) to prevent accidental conflicts with already existing identifiers.

That being said: None of the currently existing code is in any way performance critical.

Long story short: If you want to unscope the code, no biggy. It's only stylistic.

**Comment by reteo** [comment]
Fair enough.  I think I'll leave it alone for now, just checking.
